File: src/sql/postgres/protocol/ParameterStatus.zig
const ParameterStatus = @This();

name: Data = .{ .empty = {} },
value: Data = .{ .empty = {} },

pub fn deinit(this: *@This()) void {
    this.name.deinit();
    this.value.deinit();
}

pub fn decodeInternal(this: *@This(), comptime Container: type, reader: NewReader(Container)) !void {
    const length = try reader.length();
    bun.assert(length >= 4);

    this.* = .{
        .name = try reader.readZ(),
        .value = try reader.readZ(),
    };
}

pub const decode = DecoderWrap(ParameterStatus, decodeInternal).decode;

const bun = @import("bun");
const Data = @import("../../shared/Data.zig").Data;
const DecoderWrap = @import("./DecoderWrap.zig").DecoderWrap;
const NewReader = @import("./NewReader.zig").NewReader;

File: src/sql/postgres/protocol/ParameterStatus.rs
use super::new_reader::NewReader;
use crate::postgres::AnyPostgresError;
use crate::shared::Data;

#[derive(Default)]
pub struct ParameterStatus {
    pub name: Data,
    pub value: Data,
}

// The fields drop automatically, so no explicit `impl Drop` is needed.

impl ParameterStatus {
    pub fn decode_internal<Container: super::new_reader::ReaderContext>(
        mut reader: NewReader<Container>,
    ) -> Result<Self, AnyPostgresError> {
        let mut remaining = reader.body_length()?;

        let (name, consumed) = reader.string_within(remaining)?;
        remaining -= consumed;
        let (value, _) = reader.string_within(remaining)?;
        Ok(Self { name, value })
    }
}
